presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -49,6 +68,33 @@ public QuerySessionSupplier(
     }
 
     @Override
+    public void addSessionPropertyConfigurationManager(SessionPropertyConfigurationManagerFactory sessionConfigFactory)
+    {
+        if (sessionPropertyConfigurationManagerFactories.putIfAbsent(sessionConfigFactory.getName(), sessionConfigFactory) != null) {
+            throw new IllegalArgumentException(format("Session property configuration manager '%s' is already registered", sessionConfigFactory.getName()));
+        }
+    }
+
+    @Override
+    public void loadSessionPropertyConfigurationManager()
+            throws Exception
+    {
+        if (SESSION_PROPERTY_CONFIGURATION.exists()) {
+            Properties properties = new Properties();
+            try (FileInputStream in = new FileInputStream(SESSION_PROPERTY_CONFIGURATION)) {
+                properties.load(in);
+            }
+            Map<String, String> propertyMap = new HashMap<>(fromProperties(properties));
+            String configManagerName = propertyMap.remove(SESSION_PROPERTY_MANAGER_NAME);
```

Comment:
This is hard to read, add some line breaks. Also, add logging like we have in `InternalResourceGroupManager`. The logging is needed because the `create()` method can do a lot and also log stuff.
```java
log.info("-- Loading session property configuration manager --");

String configManagerName = propertyMap.remove(SESSION_PROPERTY_MANAGER_NAME);
checkArgument(configManagerName != null, "Session property configuration %s does not contain %s", SESSION_PROPERTY_CONFIGURATION, SESSION_PROPERTY_MANAGER_NAME);

SessionPropertyConfigurationManagerFactory factory = sessionPropertyConfigurationManagerFactories.get(configManagerName);
checkState(factory != null, "Session property configuration manager %s is not registered");

SessionPropertyConfigurationManager manager = factory.create(propertyMap);
checkState(sessionPropertyConfigurationManager.compareAndSet(null, manager), "sessionPropertyConfigurationManager is already set");

log.info("-- Loaded session property configuration manager %s --", configManagerName);